### PR TITLE
GitHub sync

### PR DESCRIPTION
**Pull request by nshahan**
This sync includes recent changes to the gallery generator. It is being refactored to make it easier to add new features.

### CODE REVIEW

File: example/material_stepper_example/lib/material_stepper_demo.html [hunk 1]
```diff
@@ -0,0 +1,202 @@
+<!--
+ Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+ for details. All rights reserved. Use of this source code is governed by a
+ BSD-style license that can be found in the LICENSE file.
+-->
+<section>
+  <h2>ACX2 Stepper: horizontal orientation, backwards jumps allowed, default size</h2>
```
Comment (chalin) [inline example/material_stepper_example/lib/material_stepper_demo.html]: ACX2?

File: example/material_stepper_example/lib/material_stepper_demo.html [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    <template step name="Step D Title">
+      <div class="step-d">
+        <h1>Content C</h1>
+      </div>
+    </template>
+  </material-stepper>
+</section>
+
+<section>
+  <h2>ACX2 Stepper: horizontal orientation, all jumps allowed, mini size</h2>
```
Comment (chalin) [inline example/material_stepper_example/lib/material_stepper_demo.html]: ACX2? (Here and other locations in this file.)

File: example/material_stepper_example/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+name: material_stepper_example
+author: Dart Team <misc@dartlang.org>
+environment:
+  sdk: '>=2.0.0-dev.56.0 <2.0.0'
+dependencies:
+  angular: 5.0.0-alpha+14
+  angular_components: 0.9.0-alpha+14
+  angular_gallery_section:
+    path: ../../angular_gallery_section
+  build_config: ^0.2.6
```
Comment (chalin) [inline example/material_stepper_example/pubspec.yaml]: Here and in other `pubspec.yaml` files you might want to upgrade to `build_config: ^0.3.0` (which, I believe, offers better error reporting).

File: example/material_stepper_example/lib/material_stepper_demo.html (hunk 1 above)
Comment (nshahan) [reply example/material_stepper_example/lib/material_stepper_demo.html]: Ha, not a very important description. I'll get those cleaned up.

File: example/material_stepper_example/pubspec.yaml (hunk 3 above)
Comment (nshahan) [reply example/material_stepper_example/pubspec.yaml]: Thanks. I'll look into making sure we get updated to the latest.

**Pull request merged**